### PR TITLE
Version 0.3

### PR DESCRIPTION
**Pull request by pratikyadav**
This PR

- Upgrades helper.html and index.html to GL JS v 2.7.0
- Adds a story option to change projection of map.
- Adds a story option to add a dynamic inset map.
- ~Enables `preload` for map transition.~
- Adds a default map icon for the page.
- Adds examples of some famous stories made using this template.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -252,9 +268,27 @@
...
+    style: 'mapbox://styles/mapbox/dark-v10', //hosted style id
+    center: config.chapters[0].location.center,
+    // Hardcode above center value if you want insetMap to be static.
+    zoom: 3, // starting zoom
+    hash: false,
+    interactive: false,
+    attributionControl: false,
+    //Future: Once official mapbox-gl-js has globe view enabled,
+    //insetmap can be a globe with the following parameter.
+    //projection: 'globe'
```
Comment (pratikyadav) [inline src/index.html:288]: https://user-images.githubusercontent.com/12744420/155816470-d03d6a67-8783-4270-9ef5-31f759ad82e5.mov

While the globe view right now is not available in the official release, I tested it out with a local build and it looks great.

File: src/index.html [hunk 2]
```diff
@@ -286,17 +321,34 @@
...
         offset: 0.5,
         progress: true
     })
-    .onStepEnter(response => {
+    .onStepEnter(async response => {
         var chapter = config.chapters.find(chap => chap.id === response.element.id);
         response.element.classList.add('active');
-        map[chapter.mapAnimation || 'flyTo'](chapter.location);
+        map[chapter.mapAnimation || 'flyTo'](
+            {...chapter.location, preload: true}
```
Comment (pratikyadav) [inline src/index.html:339]: Ref https://github.com/mapbox/mapbox-gl-js/pull/11328

File: src/config.js.template [hunk 3]
```diff
@@ -3,12 +3,16 @@ var config = {
     accessToken: 'YOUR_ACCESS_TOKEN',
     showMarkers: true,
     markerColor: '#3FB1CE',
-    theme: 'light',
-    use3dTerrain: false,
+    //projection: 'equirectangular',
```
Comment (willymaps) [inline src/config.js.template:6]: Should this not be commented out for visibility? Use the default `mercator`?

File: src/config.js.template [hunk 4]
```diff
@@ -3,12 +3,16 @@ var config = {
...
+    //Read more about available projections here
+    //https://docs.mapbox.com/mapbox-gl-js/example/projections/
+    inset: true,
+    theme: 'dark',
+    use3dTerrain: false, //set true for enabling 3D maps.
     title: 'The Title Text of this Story',
     subtitle: 'A descriptive and interesting subtitle to draw in the reader',
     byline: 'By a Digital Storyteller',
-    footer: 'Source: source citations, etc.',
+    footer: 'Source: source citations, etc. <br> Created using <a href="https://github.com/mapbox/storytelling">Mapbox Storytelling</a> template.',
```
Comment (willymaps) [inline src/config.js.template:15]: Might make sense to link to a new tab with `target="_blank"` in the `<a>` element

File: src/index.html (hunk 2 above)
Comment (willymaps) [reply src/index.html:339]: I'm still noticing unloaded tiles in the template:

https://user-images.githubusercontent.com/12721145/157057547-6dae2456-aff5-440d-9f7a-2da0a50c89b2.mp4

Comment (pratikyadav) [reply src/index.html:339]:
```
map[chapter.mapAnimation || 'flyTo'](
            {...chapter.location, preloadOnly: true}
        );
await map.once('idle');
map[chapter.mapAnimation || 'flyTo'](
            {...chapter.location}
        );
```



I was able to get it working with the `preloadOnly` flag, but I am now a little skeptical of it. For a style like `satellite-street`, this flag adds a significant lag as the map waits for entire tiles in the path to be loaded before starting the transition. 

I think I am going to drop this flag for now and maybe add it when the flag goes through a few more version iterations.

**Pull request merged**